extensions/wikia/AchievementsII/services/AchRankingService.class.php

```diff
@@ -163,8 +163,12 @@ public function getRecentAwardedBadges($badgeLevel = null, $limit = null, $daySp
 		while(($row = $dbr->fetchObject($res)) && (count($badges) <= $limit)) {
 			$user = User::newFromId($row->user_id);
 
-			if( $user && AchAwardingService::canEarnBadges( $user ) ) {
-				$badges[] = array('badge' => new AchBadge($row->badge_type_id, $row->badge_lap, $row->badge_level), 'date' => $row->date);
+			if ( $user && AchAwardingService::canEarnBadges( $user ) ) {
+				$badges[] = [
+					'user' => $this->extractNeededUserData( $user ),
```

Comment:
👍 
one minor comment: extractNeededUserData -> what is `needed` here?
extractUsernameAndProfileUrl or getUsernameAndProfileUrl or getPublicUserData
naming is the most challenge for devs 😉 

If those data are in many places, you can add method to user class. GetPublicData() or something.

FYI it's possible to use object again, but I don't recommended that.
@themech add JsonSerializable interface to user.
https://github.com/Wikia/app/blob/dev/includes/User.php#L4742
so only specific atributes are expose in json.